src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -49,23 +53,30 @@ private void Dynamic_All(TraceEvent obj)
                 // If it's not a counter we asked for, ignore it.
                 if (!filter.Filter(obj.ProviderName, payloadFields["Name"].ToString())) return;
 
-                // There really isn't a great way to tell whether an EventCounter payload is an instance of 
-                // IncrementingCounterPayload or CounterPayload, so here we check the number of fields 
-                // to distinguish the two.
-                ICounterPayload payload;
-                if (payloadFields.ContainsKey("CounterType"))
-                {
-                    payload = payloadFields["CounterType"].Equals("Sum") ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
-                }
-                else
+                ICounterPayload payload = payloadFields["CounterType"].Equals("Sum") ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
+                writer.Update(obj.ProviderName, payload, pauseCmdSet);
+            }
+        }
+
+        // Writes out the counter data as a user-specified file format.
+        private void DynamicAllExport(TraceEvent obj)
+        {
+            if (obj.EventName.Equals("EventCounters"))
+            {
+                IDictionary<string, object> payloadVal = (IDictionary<string, object>)(obj.PayloadValue(0));
+                IDictionary<string, object> payloadFields = (IDictionary<string, object>)(payloadVal["Payload"]);
+                if (obj.EventName.Equals("EventCounters"))
                 {
-                    payload = payloadFields.Count == 6 ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
+                    // If it's not a counter we asked for, ignore it.
+                    if (!filter.Filter(obj.ProviderName, payloadFields["Name"].ToString())) return;
+
+                    ICounterPayload payload = payloadFields["CounterType"].Equals("Sum") ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
+                    exporter.Write(obj.ProviderName, payload);
                 }
-                writer.Update(obj.ProviderName, payload, pauseCmdSet);
             }
         }
 
-        private void StopMonitor()
+        private void StopMonitor(bool liveMonitor)
```

Comment:
Assuming an interface like above this argument wouldn't be needed, you would call renderer.Stop() in place of exporter.Flush(). The file format renderers would flush in their implementation and the ConsoleWriter would do nothing.